pkg/sync/kubernetes/kubernetes_sync.go

```diff
@@ -220,48 +224,52 @@ func (k *Sync) notify(ctx context.Context, c chan<- INotify) {
 			EventType: DefaultEventTypeReady,
 		},
 	}
-	informer.Run(ctx.Done())
+
+	k.informer.Run(ctx.Done())
 }
 
-func createFuncHandler(obj interface{}, object client.ObjectKey, c chan<- INotify) error {
-	var ffObj v1alpha1.FeatureFlagConfiguration
-	u := obj.(*unstructured.Unstructured)
-	err := runtime.DefaultUnstructuredConverter.FromUnstructured(u.Object, &ffObj)
+// commonHandler emits the desired event if and only if handler receive an object matching apiVersion and resource name
+func commonHandler(obj interface{}, object client.ObjectKey, emitEvent DefaultEventType, c chan<- INotify) error {
+	ffObj, err := toFFCfg(obj)
 	if err != nil {
 		return err
 	}
-	if ffObj.APIVersion != fmt.Sprintf("%s/%s", v1alpha1.GroupVersion.Group, v1alpha1.GroupVersion.Version) {
-		return errors.New("invalid api version")
+
+	if ffObj.APIVersion != apiVersion {
+		return fmt.Errorf("invalid api version %s, expected %s", ffObj.APIVersion, apiVersion)
 	}
+
 	if ffObj.Name == object.Name {
 		c <- &Notifier{
 			Event: Event[DefaultEventType]{
-				EventType: DefaultEventTypeCreate,
+				EventType: emitEvent,
 			},
 		}
 	}
+
 	return nil
 }
 
+// updateFuncHandler handles updates. Event is emitted if and only if resource name, apiVersion of old & new are equal
 func updateFuncHandler(oldObj interface{}, newObj interface{}, object client.ObjectKey, c chan<- INotify) error {
-	var ffOldObj v1alpha1.FeatureFlagConfiguration
-	u := oldObj.(*unstructured.Unstructured)
-	err := runtime.DefaultUnstructuredConverter.FromUnstructured(u.Object, &ffOldObj)
+	ffOldObj, err := toFFCfg(oldObj)
 	if err != nil {
 		return err
 	}
-	if ffOldObj.APIVersion != fmt.Sprintf("%s/%s", v1alpha1.GroupVersion.Group, v1alpha1.GroupVersion.Version) {
-		return errors.New("invalid api version")
+
+	if ffOldObj.APIVersion != apiVersion {
+		return fmt.Errorf("invalid api version %s, expected %s", ffOldObj.APIVersion, apiVersion)
```

Comment:
can duplicate `"Simple error - API version mismatch"` test and change the api version in `validFFCfgOld` to cover this line